src/sentry/data_export/endpoints/data_export_details.py

```diff
@@ -25,6 +28,12 @@ def get(self, request, organization, **kwargs):
 
         try:
             data_export = ExportedData.objects.get(id=kwargs["data_export_id"])
+            # Check data export permissions
+            if data_export.query_info.get("project"):
+                project_ids = map(int, data_export.query_info.get("project", []))
+                projects = Project.objects.filter(organization=organization, id__in=project_ids)
+                if any(p for p in projects if not request.access.has_project_access(p)):
+                    raise PermissionDenied
```

Comment:
If we've added the UI for this, I feel like it would make sense to add a more descriptive message, something like

```python
PermissionDenied(detail="You don't have access to some of the data this export contains.")
```

Reply:
Oh damn, that's way nicer I didn't think of that!